### PR TITLE
chore: Table(Preview) Styles Refactor

### PR DESCRIPTION
**Pull request by josh-bagwell**
## Summary

Fixes: #2401  

Updates styles to use new style utilities and variables.

## Release Category
Components

### Release Note
Table now uses Canvas Tokens and our new styling utilities. The component now supports the cs prop, but otherwise the API has not changed. It should behave identically as it did in previous versions however, there may be some slight visual changes.

### BREAKING CHANGES
There may be slight visual changes.

---

## For the Reviewer

- [ ] PR title is short and descriptive
- [ ] PR summary describes the change (Fixes/Resolves linked correctly)
- [ ] PR Release Notes describes additional information useful to call out in a release message or removed if not applicable
- [ ] Breaking Changes provides useful information to upgrade to this code or removed if not applicable

## Where Should the Reviewer Start?

`/modules/preview-react/table/lib/Table.tsx`
`/modules/preview-react/table/lib/TableBody.tsx`
`/modules/preview-react/table/lib/TableCaption.tsx`
`/modules/preview-react/table/lib/TableCell.tsx`
`/modules/preview-react/table/lib/TableFooter.tsx`
`/modules/preview-react/table/lib/TableHead.tsx`
`/modules/preview-react/table/lib/TableHeader.tsx`
`/modules/preview-react/table/lib/TableRow.tsx`

Ron saying thank you

### CODE REVIEW

File: modules/preview-react/table/lib/TableHead.tsx [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
+  display: 'grid',
+  backgroundColor: base.soap100,
+  minHeight: '48px',
+  'th ': {
+    '&:first-of-type': {
+      borderLeft: 'none',
+    },
+    '&:last-of-type': {
+      borderRight: 'none',
+    },
```
Comment (RayRedGoose) [inline modules/preview-react/table/lib/TableHead.tsx:18]: Can id be: `th:where(:first-of-type, :last-of-type)`?

File: modules/preview-react/table/lib/TableHead.tsx [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
-const StyledHead = styled(Grid.as('thead'))<StyledType>({
-  'th:last-of-type': {
-    borderRight: 'none',
-  },
-  'th:first-of-type': {
-    borderLeft: 'none',
+const headStyles = createStyles({
+  display: 'grid',
+  backgroundColor: base.soap100,
+  minHeight: '48px',
```
Comment (RayRedGoose) [inline modules/preview-react/table/lib/TableHead.tsx:10]: Should be rem value

File: modules/preview-react/table/lib/TableBody.tsx [hunk 3]
```diff
@@ -1,35 +1,39 @@
...
-  },
-  'td:first-of-type': {
-    borderLeft: 'none',
+  'td ': {
+    '&:last-of-type': {
+      borderRight: 'none',
+    },
+    '&:first-of-type': {
+      borderLeft: 'none',
+    },
```
Comment (RayRedGoose) [inline modules/preview-react/table/lib/TableBody.tsx:27]: What about: `td:where(:first-of-type,:last-of-type)`?

File: modules/preview-react/table/lib/TableCaption.tsx [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
-import {colors} from '@workday/canvas-kit-react/tokens';
-import {FlexProps, Flex} from '@workday/canvas-kit-react/layout';
+import {FlexProps, mergeStyles} from '@workday/canvas-kit-react/layout';
 import {createComponent} from '@workday/canvas-kit-react/common';
+import {createStyles} from '@workday/canvas-kit-styling';
+import {base, system} from '@workday/canvas-tokens-web';
+
+const captionStyles = createStyles({
+  display: 'flex',
+  borderBottom: `1px solid ${base.soap500}`,
```
Comment (RayRedGoose) [inline modules/preview-react/table/lib/TableCaption.tsx:9]:
```suggestion
  borderBottom: `{px2rem(1)} solid ${base.soap500}`,
```

File: modules/preview-react/table/lib/TableCell.tsx [hunk 5]
```diff
@@ -1,28 +1,28 @@
...
+import {GridProps, mergeStyles} from '@workday/canvas-kit-react/layout';
 import {createComponent} from '@workday/canvas-kit-react/common';
+import {createStyles} from '@workday/canvas-kit-styling';
+import {base, system} from '@workday/canvas-tokens-web';
+
+const cellStyles = createStyles({
+  display: 'grid',
+  alignItems: 'center',
+  backgroundColor: base.frenchVanilla100,
+  borderBottom: `1px solid ${base.soap400}`,
```
Comment (RayRedGoose) [inline modules/preview-react/table/lib/TableCell.tsx:11]: `px2rem(1)`

File: modules/preview-react/table/lib/Table.tsx [hunk 6]
```diff
@@ -9,6 +8,22 @@ import {TableHeader} from './TableHeader';
...
+
+const tableStyles = createStyles({
+  display: 'grid',
+  border: `1px solid ${base.soap500}`,
+  borderRadius: system.shape.x2,
+  overflow: 'auto',
+  fontFamily: system.fontFamily.default,
+  fontSize: system.fontSize.subtext.large,
+  lineHeight: system.lineHeight.subtext.large,
+  fontWeight: system.fontWeight.normal,
```
Comment (RayRedGoose) [inline modules/preview-react/table/lib/Table.tsx:22]: Why dont you just spreading `type.levels.subtext.large`?

Comment (josh-bagwell) [reply modules/preview-react/table/lib/Table.tsx:22]: Yup!

File: modules/preview-react/table/lib/TableBody.tsx [hunk 7]
```diff
@@ -1,35 +1,40 @@
...
-  },
-  'td:first-of-type': {
-    borderLeft: 'none',
+  'td ': {
+    '&:last-of-type': {
+      borderInlineEnd: 'none',
+    },
+    '&:first-of-type': {
+      borderInlineStart: 'none',
+    },
```
Comment (mannycarrera4) [inline modules/preview-react/table/lib/TableBody.tsx:27]: you need boxSizing here ?

Comment (josh-bagwell) [reply modules/preview-react/table/lib/TableBody.tsx:27]: Yup!

**Review by mannycarrera4 — APPROVED** [review]

**Review by RayRedGoose — APPROVED** [review]
:shipit:

**Pull request merged**